File: javaspringservice/src/main/java/com/example/javaspringservice/models/entities/Message.java
package com.example.javaspringservice.models.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "message_entity")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(length = 500, nullable = false)
    private String message;
    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @ManyToOne
    @JoinColumn(name = "id_user_send")
    @JsonIgnore
    private Users usersend;

    @ManyToOne
    @JoinColumn(name = "id_user_recive")
    @JsonIgnore
    private Users userrecive;

    @PrePersist
    public void prepersist(){
        createdAt = LocalDateTime.now();
    }
}

File: typesservice/src/persistence/entities/Message.ts
import {Entity, Column, PrimaryGeneratedColumn, CreateDateColumn, ManyToOne, JoinColumn} from 'typeorm';
import { Users } from './Users';

@Entity()
export class MessageEntity{
    @PrimaryGeneratedColumn('increment')
    id:number;
    @Column({length:500, nullable:false})
    message:string;
    @CreateDateColumn({name:"created_at"})
    createdAt:Date;
    @ManyToOne(()=> Users, user => user.messagessend, {onDelete:'CASCADE'})
    @JoinColumn({name:'id_user_send'})
    usersend:Users;
    @ManyToOne(()=> Users, user => user.messagesrecive)
    @JoinColumn({name:'id_user_recive'})
    userrecive:Users;
}